toolchain/check/eval.cpp

```diff
@@ -4,31 +4,82 @@
 
 #include "toolchain/check/eval.h"
 
+#include "toolchain/sem_ir/ids.h"
 #include "toolchain/sem_ir/typed_insts.h"
 #include "toolchain/sem_ir/value_stores.h"
 
 namespace Carbon::Check {
 
+namespace {
+// The evaluation phase for an expression, computed by evaluation. These are
+// ordered so that the phase of an expression is the numerically highest phase
+// of its constituent evaluations.
+enum class Phase : uint8_t {
+  // Value could be entirely and concretely computed.
+  Template,
+  // Evaluation encountered a reference to a symbolic binding.
+  Symbolic,
+  // Evaluation encountered an error, which is treated as being potentially
+  // constant, but with an unknown phase.
+  Error,
```

Comment:
Why is Error an earlier phase than Runtime, or after Symbolic even though it carries from Template? Is this worth some comments to explain the intended interaction with LatestPhase (which I assume is why)?

Reply:
Expanded the comments to try to make this clearer.